File: apps/py/example_UnaryToUnary.py
from core.lib.decorator import channel
from core.lib.type import Request


@channel.uu
def upper(request: Request):
    return {
        "message": {
            "upper": request.message.get("raw1").upper()
        }
    }


@channel.uu
def lower(request: Request):
    return {
        "message": {
            "lower": request.message.get("raw").lower()
        }
    }

File: apps/js/example_UnaryToUnary.js
import { client } from "../../core/client/client.js";


export const rule = {
  upper: {
    reg: "^#upper",
    priority: 800,
    describe: "返回大写",
  },
  lower: {
    reg: "^#lower",
    priority: 800,
    describe: "返回小写",
  },
};

export async function upper(e) {
  client.UnaryToUnary({
    file: "example_UnaryToUnary",
    function: "upper",
    message: {
      raw: e.msg.replace("#upper", ""),
    },
  }, (err, response) => {
    if (err) {
      console.error("Error: ", err);
    } else {
      e.reply(response.message.upper);
    }
  });
}

export async function lower(e) {
  client.UnaryToUnary({
    file: "example_UnaryToUnary",
    function: "lower",
    message: {
      raw: e.msg.replace("#lower", ""),
    },
  }, (err, response) => {
    if (err) {
      console.error("Error: ", err);
    } else {
      e.reply(response.message.lower);
    }
  });
}